ha/core/controllers/pcs/pcs_controller.py

```diff
@@ -24,4 +32,80 @@ def __init__(self):
         """
         Initalize pcs controller
         """
-        super(PcsController, self).__init__()
\ No newline at end of file
+        super(PcsController, self).__init__()
+        self._execute = SimpleCommand()
+
+    def heal_resource(self, node_id):
+        """
+        Heal the resources if there are any fail count exists
+        """
+        count = 0
+        resources_healed = False
+        while True:
+            if count >= const.RETRY_COUNT:
+                break
+            fail_count_exists = self.check_resource_failcount(node_id)
+            if fail_count_exists:
+                self.clean_failure_count(node_id)
+            else:
+                resources_healed = True
+                break
+            count += 1
+            time.sleep(10)
+        return resources_healed
+
+    def check_resource_failcount(self, node_id) -> bool:
+        """
+        Resource fail count check
+        """
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_NODE_FAILCOUNT_STATUS.replace("<node>", node_id),
+                                                   check_error=False)
+        if const.NO_FAILCOUNT in _output:
+            return False
+        else:
+            return True
+
+    def clean_failure_count(self, node_id):
+        """
+        Cleanup resources fail count
+        """
+        _output, _err, _rc = self._execute.run_cmd(const.PCS_NODE_CLEANUP.replace("<node>", node_id),
+                                                   check_error=False)
```

Comment:
What if this cleanup fails ?

Reply:
It will return output and we  will check for resource fail_count again.